v2/app/incubator/raspberrypi/basic_all.py

```diff
@@ -214,21 +214,25 @@ def main():
     sensor_time = (time.time() - SENSOR_FREQ_SECONDS)
     while True:
         adjust_lights()
-        camera_delta = (time.time() - camera_time)
-        sensor_delta = (time.time() - sensor_time)
         # take picture if necessary
-        if (camera_delta > CAMERA_FREQ_SECONDS):
-            camera_time += CAMERA_FREQ_SECONDS
+        camera_delta = (time.time() - camera_time)
+        if ((camera_delta > CAMERA_FREQ_SECONDS) or (camera_retry > 0 and camera_retry < max_retry)):
             if is_lights_on():
-                took_pic = take_picture()
-                if not took_pic:
-                    camera_time = (time.time() - CAMERA_FREQ_SECONDS)
+                if take_picture():
+                    camera_time, camera_retry = (time.time(), 0)
+                else:
+                    camera_retry += 1
+                if camera_retry >= max_retry:
+                    camera_retry = 0
```

Comment:
I think we need to increment camera_time here because we will only be at max_retry if we just had a failure (camera_time not incremented) and we don't want to try to take a picture again, we want to wait for another interval to pass.